src/pages/index.js

```diff
@@ -63,27 +102,110 @@ function handleImageClick(cardLink, cardCaption) {
   imagePopup.open(cardLink, cardCaption);
 }
 
+function handleAvatarClick() {
+  editAvatarPopup.open();
+}
+
+function handleDeletePopup(cardId, evt) {
+  deleteVerifyPopup.open();
+  cardIdForDeletion = cardId;
+  cardToDelete = evt.target.closest(".card");
+}
+
+function likeToServer(cardId, cardElement) {
+  if (!cardElement.querySelector(".card__liked_active")) {
+    api.removeLikeFromCard(cardId).catch((err) => console.error(err));
+  } else {
+    api.addLikeToCard(cardId).catch((err) => console.error(err));
+  }
+}
+
 function createCard(imageInfo) {
-  const newCard = new Card(imageInfo, "#card-template", handleImageClick);
+  const newCard = new Card(
+    imageInfo,
+    "#card-template",
+    handleImageClick,
+    handleDeletePopup,
+    likeToServer
+  );
   const cardElement = newCard.generateCard();
   return cardElement;
 }
 
+function renderLoading(isLoading, button) {
+  if (isLoading) {
+    buttonText = button.textContent;
+    button.textContent = "Saving...";
+  } else {
+    button.textContent = buttonText;
+  }
+}
+
 function handleAddImage(imageInfo) {
-  const addedCardElement = createCard(imageInfo);
-  cardSection.addItem(addedCardElement, "prepend");
-  addFormPopup.close();
-  addFormPopup.clearForm();
-  formValidators["add"].disableSubmitButton();
+  renderLoading(true, addPopup.querySelector(config.submitButtonSelector));
+  api
+    .addCard(imageInfo)
+    .then((res) => {
+      let addedCardElement = createCard(res);
```

Comment:
If a variable is declared using `let`, it must be **reassigned** somewhere.
If a variable doesn’t change (is not reassigned), it is declared using `const`.

**All your variables** should be declared using `const`

Reply:
If I declare classes in functions, I can't use them outside of the functions. If I declare them as constants outside of functions, I can't change their assignments. I'm not sure what to do here.

Reply:
I mean only this `let addedCardElement = ` should be with `const`

Reply:
Ah, ok, I misunderstood then. Thanks!